### PR TITLE
Add code test coverage report

### PR DESCRIPTION
**Pull request by dracco1993**

### CODE REVIEW

**Comment by theblindbandet** [comment]
wait, why are we adding back in the config stuff from jaws? that was literally just a badly written implimentation for making the porting from C++ easier.

File: .github/workflows/code-coverage.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# This workflow will build a Java project with Gradle
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-gradle
+
+name: Java CI with Gradle
```
Comment (theblindbandet) [inline .github/workflows/code-coverage.yml:4]: Probably want to rename this to CodeCoverage or something.

**Review by theblindbandet — APPROVED** [review]
the_user_actually_approves

**Review by theblindbandet — APPROVED** [review]

**Review by tricktrap — APPROVED** [review]
Many changes, looks good

**Pull request merged**